pcdsdevices/attenuator.py

```diff
@@ -441,3 +443,245 @@ class GasAttenuator(Device, BaseInterface):
 
     not_implemented = Cpt(SignalRO, name="Not Implemented",
                           value="Not Implemented", kind='normal')
+
+
+class AttenuatorCalculatorFilter(Device, BaseInterface):
+    material = Cpt(
+        EpicsSignal, 'Material', kind='hinted', string=True,
+        doc='The material formula (e.g., Si, C)'
+    )
+    thickness = Cpt(
+        EpicsSignal, 'Thickness', kind='hinted',
+        doc='Thickness in micron',
+    )
+    is_stuck = Cpt(
+        EpicsSignal, 'IsStuck', kind='hinted',
+        doc='Is the filter stuck / unusable?',
+    )
+    closest_energy = Cpt(
+        EpicsSignalRO, 'ClosestEnergy_RBV', kind='config',
+        doc='Closest tabulated energy available to the requested one',
+    )
+    transmission = Cpt(EpicsSignal, 'Transmission_RBV', kind='normal',
```

Comment:
it's kind of cool you're doing this per-filter, the previous calculation IOCs didn't have this